pilot/pkg/xds/ads.go

```diff
@@ -323,6 +327,92 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 	}
 }
 
+// StreamSecrets serves SDS discovery requests and SDS push requests
+func (s *DiscoveryServer) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecretsServer) error {
```

Comment:
maybe better to move to a separate sds file

Reply:
Can do that - short term since the code is almost identical with the one for ADS it may be good to keep the next to each other, so people don't forget to keep them in sync. Most of it should be boilerplate, due to gRPC limitations - both should have the same Generator logic.

Reply:
Can't it be literally identical aside from the function name?

Reply:
the 'stream' has different type. I tried adding an interface with 2 implementations - that works, but seems a bit more intrusive and risky. It made sense when the code was super complex - but now the method is pretty small and simple.

Reply:
We have an abstraction though, DiscoveryStream. So I would expect we have 

```go
func serveStream(stream DiscoveryStream) {
 ... all logic today ...
}

func StreamAggregatedResources(stream discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer) {
return  serveStream(stream)
}
func ServeSecret(stream sds.SecretDiscoveryService_StreamSecretsServer) {
return  serveStream(stream)
}
```

would work

Reply:
Didn't see that - I did a similar thing long ago. Cool - changed, thanks for the tip.